packages/statemanager/src/ethersStateManager.ts

```diff
@@ -0,0 +1,382 @@
+
+import { Cache } from './cache'
+
+import { BaseStateManager } from '.'
+
+import type { Proof, StateManager } from '.'
+import type { getCb, putCb } from './cache'
+import type { StorageDump } from './interface'
+import type { Common } from '@ethereumjs/common'
+import type { Address } from '@ethereumjs/util'
+
+const log = debug('statemanager')
+
+type BlockTagType = bigint | 'latest' | 'earliest' | 'pending'
+export interface EthersStateManagerOpts {
+  provider: string | StaticJsonRpcProvider | JsonRpcProvider
+  blockTag?: BlockTagType
+}
+
+export class EthersStateManager extends BaseStateManager implements StateManager {
+  private provider: StaticJsonRpcProvider | JsonRpcProvider
+  private contractCache: Map<string, Buffer>
+  private storageCache: Map<string, Map<string, Buffer>>
+  private blockTag: string
+  private stateRoot: Buffer | undefined
+  _cache: Cache
+
+  constructor(opts: EthersStateManagerOpts) {
+    super({})
+    if (typeof opts.provider === 'string') {
+      this.provider = new StaticJsonRpcProvider(opts.provider)
+    } else if (opts.provider instanceof JsonRpcProvider) {
+      this.provider = opts.provider
+    } else {
+      throw new Error(`valid JsonRpcProvider or url required; got ${opts.provider}`)
+    }
+
+    this.blockTag =
+      typeof opts.blockTag === 'bigint' ? bigIntToHex(opts.blockTag) : opts.blockTag ?? 'latest'
+
+    this.contractCache = new Map()
+    this.storageCache = new Map()
+
+    /*
+     * For a custom StateManager implementation adopt these
+     * callbacks passed to the `Cache` instantiated to perform
+     * the `get`, `put` and `delete` operations with the
+     * desired backend.
+     */
+    const getCb: getCb = async (address) => {
+      return this.getAccountFromProvider(address)
+    }
+    const putCb: putCb = async (_keyBuf, _accountRlp) => {
+      return Promise.resolve()
+    }
+    const deleteCb = async (_keyBuf: Buffer) => {
+      return Promise.resolve()
+    }
+    this._cache = new Cache({ getCb, putCb, deleteCb })
+  }
+
+  copy(): EthersStateManager {
+    const newState = new EthersStateManager({ provider: this.provider })
+    ;(newState as any).contractCache = new Map(this.contractCache)
+    ;(newState as any).storageCache = new Map(this.storageCache)
+    return newState
+  }
+
+  /**
+   * Sets the new block tag used when querying the provider and clears the
+   * internal cache.
+   * @param blockTag - the new block tag to use when querying the provider
+   */
+  setBlockTag(blockTag: BlockTagType): void {
+    if (typeof blockTag === 'bigint') {
+      this.blockTag = bigIntToHex(blockTag)
+    } else {
+      this.blockTag = blockTag
+    }
+    this.clearCache()
+  }
+
+  /**
+   * Clears the internal cache so all accounts, contract code, and storage slots will
+   * initially be retrieved from the provider
+   */
+  clearCache(): void {
+    this.contractCache.clear()
+    this.storageCache.clear()
+    this._cache.clear()
```

Comment:
Difference here: `this._cache.clear()` hasn't been in the "old" code.

Reply:
Yep, we don't have a `cache` in the trie-backed version of the PR so I added this hear to ensure the cache was cleared